pydantic/env_settings.py

```diff
@@ -27,20 +29,66 @@ def __init__(
         __pydantic_self__,
         _env_file: Union[Path, str, None] = env_file_sentinel,
         _env_file_encoding: Optional[str] = None,
+        _secrets_dir: Union[Path, str, None] = None,
         **values: Any,
     ) -> None:
         # Uses something other than `self` the first arg to allow "self" as a settable attribute
         super().__init__(
-            **__pydantic_self__._build_values(values, _env_file=_env_file, _env_file_encoding=_env_file_encoding)
+            **__pydantic_self__._build_values(
+                values, _env_file=_env_file, _env_file_encoding=_env_file_encoding, _secrets_dir=_secrets_dir
+            )
         )
 
     def _build_values(
         self,
         init_kwargs: Dict[str, Any],
         _env_file: Union[Path, str, None] = None,
         _env_file_encoding: Optional[str] = None,
+        _secrets_dir: Union[Path, str, None] = None,
     ) -> Dict[str, Any]:
-        return deep_update(self._build_environ(_env_file, _env_file_encoding), init_kwargs)
+        return deep_update(
+            self._build_secrets(_secrets_dir), self._build_environ(_env_file, _env_file_encoding), init_kwargs
+        )
+
+    def _build_secrets(self, _secrets_dir: Union[Path, str, None] = None) -> Dict[str, Optional[str]]:
+        """
+        Build secrets suitable for passing to the Model.
+        """
+        d: Dict[str, Optional[str]] = {}
+
+        secrets_dir = _secrets_dir or self.__config__.secrets_dir
+        if secrets_dir is None:
+            return d
+
+        secrets_path = Path(secrets_dir)
+
+        if not secrets_path.is_dir():
+            return d
+
+        secret_paths = {
+            (item.name if self.__config__.case_sensitive else item.name.lower()): item
+            for item in secrets_path.iterdir()
+            if item.is_file()
+        }
+
+        for field in self.__fields__.values():
+            if field.type_ not in (SecretStr, SecretBytes):
+                warnings.warn(
+                    f'found secret file matching field "{field.name}" but the field was not '
```

Comment:
wait, this is in the wrong place I think. No file has been found yet, we're just iterating through fields.